src/libs/DateUtils.js

```diff
@@ -324,6 +422,83 @@ function getStatusUntilDate(inputDate) {
     return translateLocal('statusPage.untilTime', {time: format(input, `${CONST.DATE.FNS_FORMAT_STRING} ${CONST.DATE.LOCAL_TIME_FORMAT}`)});
 }
 
+/**
+ * Update the hour and minute of a date.
+ *
+ * @param {string} inputTime - Time in "hh:mm A" format (like "10:55 AM").
+ * @param {string} inputDateTime - Date in "YYYY-MM-DD HH:mm:ss" format.
+ * @returns {string} - Date with updated time in "YYYY-MM-DD HH:mm:ss" format.
+ */
+const combineDateAndTime = (inputTime, inputDateTime) => {
+    // Parse the time from the input
+    const parsedTime = moment(inputTime, 'hh:mm A');
+
+    // Create a moment object from the input date-time (or just date)
+    const parsedDateTime = moment(inputDateTime, inputDateTime.includes(':') ? 'YYYY-MM-DD HH:mm:ss' : 'YYYY-MM-DD');
+
+    // Set the hour and minute from the time input to the date from the date-time input
+    parsedDateTime.set({
+        hour: parsedTime.get('hour'),
+        minute: parsedTime.get('minute'),
+    });
+
+    return parsedDateTime.format('YYYY-MM-DD HH:mm:ss');
+};
+
+/**
+ * @param {String} time
+ * @returns {Object}
+ * @example parseTimeTo12HourFormat('2023-08-29 11:10:00') // {hour: '11', minute: '10', period: 'AM'}
+ */
+function parseTimeTo12HourFormat(time) {
+    if (!time) {
+        return {
+            hour: '00',
+            minute: '01',
+            period: 'AM',
+        };
+    }
+    const parsedTime = moment(time, 'h:mm A');
+
+    return {
+        hour: parsedTime.format('hh'), // Hour in 12-hour format
+        minute: parsedTime.format('mm'), // Minute
+        period: parsedTime.format('A'), // AM or PM
+    };
+}
+
+/**
+ * @param {String} dateStr
```

Comment:
Also unsure what this one is doing exactly :smile:

Reply:
It's receive date like '11:10 PM' and return AM or PM value

Reply:
Ah so the "random date" part is just to force moment to parse it? It's a bit hacky-looking, but if there's no better way I guess we can keep it for now :smile: Since it actually accepts a time string, can you rename the input param to `timeStr`?